src/platform/posix/tcp_ip_channel.c

```diff
@@ -220,7 +232,7 @@ void TcpIpChannel_register_callback(NetworkChannel *untyped_self,
     throw("pthread_attr_init failed");
   }
 /* TODO: RIOT posix-wrappers don't have pthread_attr_setstack yet */
-#ifdef PLATFORM_RIOT
+#if defined(PLATFORM_RIOT) && !defined(__USE_XOPEN2K)
```

Comment:
When compiling for RIOT native BOARD we otherwise get a deprecated warning. Native does support the new setstack function